pulumi/infra/consul_intentions.py

```diff
@@ -36,5 +36,7 @@ def __init__(
                         kind=intention["Kind"],
                         name=intention["Name"],
                         config_json=json.dumps({"Sources": intention["Sources"]}),
-                        opts=pulumi.ResourceOptions(parent=self),
+                        opts=pulumi.ResourceOptions.merge(
```

Comment:
This merges the resource options so we can use the provider option we're passing in. That way we use the real address for consul.